client/src/js/services/reference.js

```diff
@@ -0,0 +1,34 @@
+'use strict';
+
+/**
+ * @description 
+ *
+ * @returns 
+ */
+
+angular.module('bhima.services')
+  .factory('reference', reference);
```

Comment:
Do these need to be factories?  Why not make them services (considering they are in a module "bhima.services")?